### PR TITLE
Migrate from springfox to springdoc

### PR DESCRIPTION
**Pull request by kkehl-usgs**
This pull request replaces springfox with springdoc, and updates annotations to swagger 3.

### CODE REVIEW

**Comment by kkehl-usgs** [comment]
I've followed the redirect example Dave gave me and everything looks good now.   I'm seeing the right server info and the custom title and description I added, etc.

**Comment by kkehl-usgs** [comment]
I hooked up the database and tested the APIs and it works.   I initially tried to remove "/linked-data" from ConfigurationService.getRootUrl(), but many tests failed after that and such a big change seemed out of scope for this third-party upgrade issue.

**Comment by jkreft-usgs** [comment]
I would call this less of a dependency upgrade and more of a refactor of how the project generates a key deliverable- the switch from springfox to springdoc has been a chunk of work for each project that has done it.  But we can then use openAPI3, not get dinged for javascript issues, etc so it has been worth it.  This is the last IOW java project that needed to make the swap.

File: src/main/java/gov/usgs/owi/nldi/services/ConfigurationService.java [hunk 1]
```diff
@@ -35,6 +35,10 @@ public String getRootUrl() {
 		return getDisplayProtocol() + "://" + getDisplayHost() + getDisplayPath() + "/linked-data";
 	}
 
+	public String getSwaggerUrl() {
```
Comment (dsteinich) [inline src/main/java/gov/usgs/owi/nldi/services/ConfigurationService.java:38]: I think there may be less confusion in the long run if getRootUrl() where renamed getLinkedDataUrl() and this method (getSwaggerUrl()) would be renamed getRootUrl().

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 spring:
   datasource:
-    url: "jdbc:postgresql://${nldiDbHost}:${nldiDbPort}/nldi"
+    url: "jdbc:postgresql://${nldiDbHost}:${nldiDbPort}/${nldiDbName}"
```
Comment (dsteinich) [inline src/main/resources/application.yml:3]: Need to remember to add this to the deploy configuration.

**Comment by kkehl-usgs** [comment]
Closing this pull request so I can merge from a fork rather than a remote branch.

**Pull request closed without merging**